packages/gatsby/src/internal-plugins/query-runner/query-compiler.js

```diff
@@ -66,7 +66,7 @@ const overlayErrorID = `graphql-compiler`
 
 const resolveThemes = (plugins = []) =>
   plugins.reduce((merged, plugin) => {
-    if (plugin.resolve.includes(`gatsby-theme-`)) {
+    if (plugin.name.includes(`gatsby-theme-`)) {
```

Comment:
plugin.resolve is the symlinked path when running `yarn link`. This change will work with symlinked modules and "regular" (e.g. modules in node_modules) alike.